html-aam/roles.html

```diff
@@ -0,0 +1,223 @@
+<dfn data-testname="el-dfn" data-expectedrole="term" class="ex">x</dfn>
+<div open data-testname="el-div" data-expectedrole="generic" class="ex">x</div>
+<!-- todo: dl -->
+
+<!-- el-dt -->
+<dl>
+  <dt data-testname="el-dt" data-expectedrole="term" class="ex">x</dt>
+  <dd>x</dd>
+</dl>
+
+<em data-testname="el-em" data-expectedrole="emphasis" class="ex">x</em>
+<!-- todo: embed -->
+<fieldset data-testname="el-fieldset" data-expectedrole="group" class="ex"><legend>x</legend><input></fieldset>
+<!-- todo: figcaption -->
+<figure data-testname="el-figure" data-expectedrole="figure" class="ex"><img src="#" alt="x"><figcaption>x</figcaption></figure>
+
+<!-- el-footer -->
+<nav>
+  <footer data-testname="el-footer" data-expectedrole="generic" aria-label="x" class="ex">x</aside>
+</nav>
+<footer data-testname="el-footer-ancestorbody" data-expectedrole="contentinfo" class="ex">x</footer>
+
+<form data-testname="el-form" data-expectedrole="form" class="ex"><input></form>
+<!-- todo: form-associated custom element -->
+
+<!-- el-h1-h6 -->
+<h1 data-testname="el-h1" data-expectedrole="heading" class="ex">x</h1>
+<h2 data-testname="el-h2" data-expectedrole="heading" class="ex">x</h2>
+<h3 data-testname="el-h3" data-expectedrole="heading" class="ex">x</h3>
+<h4 data-testname="el-h4" data-expectedrole="heading" class="ex">x</h4>
+<h5 data-testname="el-h5" data-expectedrole="heading" class="ex">x</h5>
+<h6 data-testname="el-h6" data-expectedrole="heading" class="ex">x</h6>
+
+<!-- skip: head -->
+
+<!-- el-header -->
+<nav>
+  <header data-testname="el-header" data-expectedrole="generic" aria-label="x" class="ex">x</header>
+</nav>
+<header data-testname="el-header-ancestorbody" data-expectedrole="banner" class="ex">x</header>
+
+<hgroup data-testname="el-hgroup" data-expectedrole="group" class="ex"><h1>x</h1></hgroup>
+<hr data-testname="el-hr" data-expectedrole="separator" class="ex">
+<!-- todo: html -->
+<i data-testname="el-i" data-expectedrole="generic" class="ex">x</i>
+<!-- todo: iframe -->
+<img src="#" alt="x" data-testname="el-img" data-expectedrole="image" class="ex">
+
+<!-- Implementations might also be valid if ignored rather than returning 'none' for the following images. -->
+<img src="#" alt data-testname="el-img-alt-no-value" data-expectedrole="none" class="ex">
+<img src="#" alt="" data-testname="el-img-empty-alt" data-expectedrole="none" class="ex">
+
+<input type="button" value="x" data-testname="el-input-button" data-expectedrole="button" class="ex">
+<input type="checkbox" data-testname="el-input-checkbox" data-expectedrole="checkbox" class="ex">
+<!-- todo: input type="color" -->
+<!-- todo: input type="date" -->
+<!-- todo: input type="datetime" -->
+<!-- todo: input type="datetime-local" -->
+<input type="email" data-testname="el-input-email" data-expectedrole="textbox" class="ex">
+<!-- todo: input type="file" -->
+<!-- skip: input type="hidden" -->
```

Comment:
do you use this "skip" language any where else? maybe instead of "skipped" we should say "no mapped" or "not specified" -- I feel like "skip" is too vague/confusing.

I like that they are in here, feels very thorough!

Reply:
changed to "tag name (not mapped)"